libraries/standard/mqtt/src/mqtt.c

```diff
@@ -1051,7 +1082,7 @@ static MQTTStatus_t sendPublish( MQTTContext_t * pContext,
                             pContext->networkBuffer.pBuffer,
                             headerSize );
 
-    if( bytesSent < 0 )
+    if( ( bytesSent < 0 ) || ( ( size_t ) bytesSent != headerSize ) )
```

Comment:
Isn't the check ( ( size_t ) bytesSent != headerSize ) just enough here?

Reply:
My reasoning here is that if bytesSent is a negative number, then casting it immediately to unsigned will lose information.